live_data_subscriber/__init__.py

```diff
@@ -22,14 +24,15 @@
 }
 
 
-def generate_topic_api_client_args_mapping():
+def generate_api_client_args_mapping():
     sim_api_domain_name = environ["API_CLIENT_SIM_API_DOMAIN_NAME"]
     external_api_domain_name = environ["API_CLIENT_DOMAIN_NAME"]
     websocket_api_domain_name = environ["API_CLIENT_WEBSOCKET_DOMAIN_NAME"]
 
     cn_mapping = list_running_canary_networks_and_devices_with_live_data(sim_api_domain_name)
 
     topic_api_client_mapping = {v: [] for _, v in allowed_devices_name_mapping.items()}
+    device_api_client_mapping = {k: [] for k, _ in allowed_devices_name_mapping.items()}
```

Comment:
so for the future, a new `allowed_devices_name_mapping` has to be created, rigth? For that we need information from the costumer, right?

Reply:
How about making this an input variable and only returning one `device_api_client_mapping` (not a tuple).
This way you can decide in `refresh_cn_and_device_list` which mapping is used.

Reply:
> so for the future, a new allowed_devices_name_mapping has to be created, rigth? For that we need information from the costumer, right?

May be, not sure, probably depends on client's requirement to either support both mqtt as well as ws or one of it for their devices, for later case having a separate `allowed_devices_name_mapping ` for mqtt and ws sounds better.

> How about making this an input variable...
Sorry didn't completely get you ,Input variable to what?

Reply:
What i meant is that you are performing the same steps for creating `device_api_client_mapping` and `topic_api_client_mapping`. The only difference in the future will be the `allowed_devices_name_mapping`, so why not making this an input variable for `generate_api_client_args_mapping`?

Reply:
Adapted via https://github.com/gridsingularity/d3a-api-client/pull/113/commits/b80f0a53a24d3d51f97f66d5b6d470c9897248f9